### PR TITLE
Add Fluent smoke test and refresh module_max_lines UI

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- add a standalone Fluent smoke-check that scans every bundle and fails fast on parse or duplicate issues
- normalise isolation-mark handling across localisation behaviour tests and update Welsh expectations
- rebuild the module_max_lines UI harness with per-fixture configs, refined span counting, and deterministic fixtures

## Testing
- make check-fmt
- make lint
- make test

------
https://chatgpt.com/codex/tasks/task_e_6909bd9d7c6c8322b44e99bc5187d35b

## Summary by Sourcery

Add a standalone Fluent smoke-test suite for parsing and duplicate detection across all localization bundles, revamp the module_max_lines UI test harness with per-fixture configs and improved span counting, normalize isolation mark handling in behaviour tests, and update Welsh localization expectations and documentation accordingly.

New Features:
- Introduce a Fluent smoke-test suite that discovers every .ftl bundle and fails fast on parse errors or duplicate message IDs using rstest-bdd scenarios

Enhancements:
- Reconstruct the module_max_lines UI harness to support per-fixture dylint.toml configs, deterministic fixture discovery, custom copying, and refined lint span calculation
- Normalize Unicode isolation mark stripping across localization behaviour tests for clearer assertions
- Improve module_max_lines logic by refining module span extraction fallback and computing line counts via first/last line indices
- Update Welsh localization expectations in BDD scenarios for lint count and conditional branches

Documentation:
- Document the new .ftl smoke tests in the design guide and adjust formatting guidelines in the rstest-bdd user guide

Tests:
- Refactor i18n test modules to use default arguments, strip isolation marks, and leverage message_with_args for consistency

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[common/tests/i18n_quality/ftl_smoke_behaviour.rs](https://github.com/leynos/whitaker/pull/34/files/bf11dd7efb78624c7719a4074c2a6679f44d3fd3#diff-5d5d1462571b4470be5dcb2c20ac847e921daffc526552ba807b092ce857c75d)

Comment on lines +63 to +87

~~~rust
fn duplicate_message_count(source: &str) -> usize {
    let mut seen: BTreeSet<String> = BTreeSet::new();
    let mut duplicates = 0;

    for line in source.lines() {
        if line.starts_with(' ') || line.starts_with('\t') {
            continue;
        }
        let trimmed = line.trim_start();
        if trimmed.starts_with('#') || trimmed.is_empty() {
            continue;
        }
        if let Some((identifier, _)) = trimmed.split_once('=') {
            let id = identifier.trim();
            if id.is_empty() {
                continue;
            }
            if !seen.insert(id.to_string()) {
                duplicates += 1;
            }
        }
    }

    duplicates
}
~~~

❌ New issue: [Bumpy Road Ahead](https://codescene.io/projects/71836/delta?repo-id=1205552&review-id=34&biomarker=Bumpy+Road+Ahead&filename=common%2Ftests%2Fi18n_quality%2Fftl_smoke_behaviour.rs&method=duplicate_message_count)
duplicate_message_count has 2 blocks with nested conditional logic. Any nesting of 2 or deeper is considered. Threshold is 2 blocks per function

File: crates/module_max_lines/src/lib.rs [hunk 1]
```diff
@@ -124,18 +123,37 @@ fn load_configuration() -> usize {
...
 
-    Some(info.lines.len())
+    let Some(first) = info.lines.first() else {
+        return None;
+    };
+    let Some(last) = info.lines.last() else {
+        return None;
+    };
+
+    Some(last.line_index.saturating_sub(first.line_index) + 1)
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/module_max_lines/src/lib.rs]: Has this now been resolved?

File: crates/module_max_lines/src/lib.rs [hunk 2]
```diff
@@ -322,32 +340,136 @@ mod behaviour {
...
+        fixtures.sort();
+
+        for source in fixtures {
+            run_fixture(crate_name, directory, &source)?;
+        }
+
+        Ok(())
+    }
+
+    fn discover_fixtures(directory: &Utf8Path) -> io::Result<Vec<PathBuf>> {
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/module_max_lines/src/lib.rs]: Has this now been resolved?

File: crates/module_max_lines/src/lib.rs [hunk 3]
```diff
@@ -322,32 +340,136 @@ mod behaviour {
...
+            let path = entry.path();
+            let target = destination.join(entry.file_name());
+            if entry.file_type()?.is_dir() {
+                copy_directory(&path, &target)?;
+            } else {
+                fs::copy(&path, &target)?;
+            }
+        }
+        Ok(())
+    }
```
[the remark this answers is not in the document]
Comment (leynos) [reply crates/module_max_lines/src/lib.rs]: Has this now been resolved?
